### DISCUSSION TITLE
Multiple files in array (Content-Type: multipart/form-data)

### DISCUSSION THREAD

**Discussion by paulbrosowsky** [dedoc/scramble#259] [Q&A] [answered]
Hello there,

im uploading multiple files using `multipart/form-data` request.
Here are my validation rules:
```
 $valideData = $request->validate([
      'files' => ['required', 'array'],
      'files.*' => ['required', 'image:jpg, png', 'max:5120'],
]);
```
It seems that binaries are not correctly detected on this level. The  documentation generates `application/json` in the header. 
Is it possible to override request headers manually?

<img width="1342" alt="Screenshot 2023-10-19 at 15 51 14" src="https://github.com/dedoc/scramble/assets/34491802/7dac738a-2373-487b-b9b7-362cd567e8ea">

**Comment by WildEgo** [comment]
> Hello there,
> 
> im uploading multiple files using `multipart/form-data` request.
> Here are my validation rules:
> ```
>  $valideData = $request->validate([
>       'files' => ['required', 'array'],
>       'files.*' => ['required', 'image:jpg, png', 'max:5120'],
> ]);
> ```
> It seems that binaries are not correctly detected on this level. The  documentation generates `application/json` in the header. 
> Is it possible to override request headers manually?
> 
> <img width="1342" alt="Screenshot 2023-10-19 at 15 51 14" src="https://github.com/dedoc/scramble/assets/34491802/7dac738a-2373-487b-b9b7-362cd567e8ea">
> 
> 

I'll try have a look at it when I'm back, in the meantime I think you can use a decorator to set it as multipart/form-data, I can't remember which but check my commit history you'll see it

**Comment by romalytvynenko** [comment, marked as answer]
@paulbrosowsky please try `@requestMediaType` out:

```php
/**
 * @requestMediaType multipart/form-data
 */
public function index(Request $request)
{
    $valideData = $request->validate([
        'files' => ['required', 'array'],
        'files.*' => ['required', 'image:jpg, png', 'max:5120'],
    ]);
}
```

**Reply by paulbrosowsky** [reply]
Yes it works, thank you!

**Reply by nerisonpitogo** [reply]
will you able to attach a file and test the endpoint? 

![image](https://github.com/user-attachments/assets/0085ab05-79d3-41c8-9619-0a0abaff92f8)

**Reply by romalytvynenko** [reply]
@nerisonpitogo sadly, Stoplight Elements (a lib Scramble uses for UI) has a bug which prevents files uploading.

**Reply by nerisongotham** [reply]
Will there be any file uploads in the future? Anyway without it, this is still super great. Will introduce this to my mates. You have a wonderful project @romalytvynenko .

**Reply by Enesi-George** [reply]
Hi, please did you copy and paste the image path directly or you click to upload from the scramble page?

**Reply by romalytvynenko** [reply]
@Enesi-George if you use the latest Scramble version, file upload should be working

**Reply by Enesi-George** [reply]
> @Enesi-George if you use the latest Scramble version, file upload should be working

@romalytvynenko mine is not working 
![image](https://github.com/user-attachments/assets/30fdeaaf-3cbd-4905-9209-ee3cfcbcddd0)
![picture_ui](https://github.com/user-attachments/assets/01389a90-230d-4da1-b6c9-e3a61d4ad0f8)

**Reply by farzadaqajani** [reply]
> @paulbrosowsky please try `@requestMediaType` out:
> 
> ```
> /**
>  * @requestMediaType multipart/form-data
>  */
> public function index(Request $request)
> {
>     $valideData = $request->validate([
>         'files' => ['required', 'array'],
>         'files.*' => ['required', 'image:jpg, png', 'max:5120'],
>     ]);
> }
> ```

This problem still exists.
